src/displayBackend/wayland/WaylandZCopy.cpp

```diff
@@ -78,8 +78,15 @@ void WaylandZCopy::onDevice(const string& name)
 
 	LOG(mLog, DEBUG) << "onDevice name: " << name;
 
-	mDrmDevice.reset(new Drm::DisplayWayland(name));
-
+    try
```

Comment:
Indentation is wrong.

Also, I don't think that logging an exception is the right way to handle it. I'm 99% sure that application will be in wrong state after that. You either need to implement proper handling logic or crash the app.

Reply:
> Indentation is wrong.
> 
> Also, I don't think that logging an exception is the right way to handle it. I'm 99% sure that application will be in wrong state after that. You either need to implement proper handling logic or crash the app.

The case here is to avoid the possible crash because of run-time issue.
The wrong state can be clarified in log, this is better than have a crash.
Saying about the algorithm of wrong state managing - yes, it must be investigated and I propose create the separate issue to start the investigation. Because problem exists.

Reply:
Now, I'm tend to disagree. Program in incorrect state will eventually crash. But it will be much harder to determine the root cause of that crash. Core dump will not have information of program state prior to the original fault.

Reply:
> Now, I'm tend to disagree. Program in incorrect state will eventually crash. But it will be much harder to determine the root cause of that crash. Core dump will not have information of program state prior to the original fault.

So, you propose to leave as is and rely at core-dump until the new algorithm will be implemented ?

Reply:
> Indentation is wrong.
> 
> Also, I don't think that logging an exception is the right way to handle it. I'm 99% sure that application will be in wrong state after that. You either need to implement proper handling logic or crash the app.

below the sample of the implementation , in the case of error - logging and break execution without the exception
static void
drm_handle_device(void *data, struct wl_drm *drm, const char *device)
{
    VADisplayContextP const pDisplayContext = data;
    VADriverContextP const ctx = pDisplayContext->pDriverContext;
    VADisplayContextWaylandDRM * const wl_drm_ctx = pDisplayContext->opaque;
    struct drm_state * const drm_state = ctx->drm_state;
    drm_magic_t magic;
    struct stat st;

    if (stat(device, &st) < 0) {
        va_wayland_error("failed to identify %s: %s (errno %d)",
                         device, strerror(errno), errno);
        return;
    }

    if (!S_ISCHR(st.st_mode)) {
        va_wayland_error("%s is not a device", device);
        return;
    }

    drm_state->fd = open(device, O_RDWR);
    if (drm_state->fd < 0) {
        va_wayland_error("failed to open %s: %s (errno %d)",
                         device, strerror(errno), errno);
        return;
    }

    drmGetMagic(drm_state->fd, &magic);
    wl_drm_authenticate(wl_drm_ctx->drm, magic);